project-metrics/metrics_service/apis/circleci.py

```diff
@@ -0,0 +1,51 @@
+"""Client for querying CircleCI API.
+
+For an overview of the API https://circleci.com/docs/api/v2/
+"""
+import json
+from typing import Dict
+from absl import logging
+import logging
+import requests
+from database import models
+
+import env
+
+CIRCLECI_API = 'https://circleci.com/api/v2'
+VCS_STAB = 'github'
+PROJECT = env.get('GITHUB_REPO')
+INSIGHTS_API = '%s/insights/%s/%s' % (CIRCLECI_API, VCS_STAB, PROJECT)
+
+
+class CircleCiError(Exception):
+  def __init__(self, status_code: int, message: str):
+    msg = 'CircleCI API Error status %d. Message: %s' % (status_code, message)
+    super(CircleCiError, self).__init__(msg)
+
+
+class CircleCiAPI(object):
+  """Interface class for executing queries agains the CircleCI API."""
+  def __init__(self, token = env.get('CIRCLECI_API_ACCESS_TOKEN')):
+    self._token = token
+
+  def _dict_to_params(self, params: Dict):
+    param_string = '&'.join(['%s=%s' % (key, params[key]) for key in params.keys()])
+    if len(param_string):
+      return '?' + param_string
+    return param_string
+
+  def get_workflow_stats(self, reporting_window = models.CircleCiReportingWindow.LAST_90_DAYS) -> models.CircleCiWorkflowStats:
+    params = {
+      'reporting-window': reporting_window.value
+    }
+    endpoint = '%s/%s' % (INSIGHTS_API, 'workflows')
```

Comment:
Consider format strings (`f'{INSIGHTS_API}/workflows'`) here and elsewhere

Reply:
I didn't know about this syntax, that definitely seems better.